Mathlib/RingTheory/MvPowerSeries/Substitution.lean

```diff
@@ -0,0 +1,1096 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+
+import Mathlib.Algebra.MvPolynomial.CommRing
+import Mathlib.LinearAlgebra.Finsupp.Pi
+import Mathlib.RingTheory.MvPowerSeries.Evaluation
+import Mathlib.RingTheory.MvPowerSeries.LinearTopology
+import Mathlib.RingTheory.MvPowerSeries.Trunc
+import Mathlib.RingTheory.PowerSeries.Basic
+import Mathlib.RingTheory.Nilpotent.Basic
+import Mathlib.Topology.Algebra.Algebra
+
+/-! # Substitutions in power series
+
+Here we define the substitution of power series into other power series.
+We follow Bourbaki, Algèbre, chap. 4, §4, n° 3.
+
+Evaluation of a power series at adequate elements has been defined
+in `DividedPowers.ForMathlib.MvPowerSeries.Evaluation.lean`.
+The goal here is to check the relevant hypotheses:
+* The ring of coefficients is endowed the discrete topology.
+* The main condition rewrites as having vanishing constant coefficient
+* Power series have a linear topology
+-/
+
+theorem Set.Finite.support_of_summable
+    {α : Type*} [AddCommGroup α] [TopologicalSpace α] [DiscreteTopology α]
+    {β : Type*} (f : β → α) (h : Summable f) : Set.Finite f.support := by
+  obtain ⟨a, ha⟩ := h
+  simp only [HasSum] at ha
+  classical
+  simp_rw [tendsto_atTop_nhds] at ha
+  obtain ⟨s, hs⟩ := ha {a} rfl (isOpen_discrete _)
+  apply Set.Finite.subset s.finite_toSet
+  intro b
+  rw [Function.mem_support, not_imp_comm]
+  intro hb
+  let hs' := hs (insert b s) (s.subset_insert b)
+  simp only [Set.mem_singleton_iff] at hs hs'
+  simpa [Finset.sum_insert hb, hs, add_left_eq_self] using hs'
```

Comment:
```suggestion
@[to_additive]
theorem Multipliable.finite_mulSupport
    {α : Type*} [CommGroup α] [TopologicalSpace α] [DiscreteTopology α]
    {β : Type*} (f : β → α) (h : Multipliable f) : Set.Finite f.mulSupport := by
  haveI : IsTopologicalGroup α := ⟨⟩
  exact h.tendsto_cofinite_one (discreteTopology_iff_singleton_mem_nhds.mp ‹_› 1)
```

This should go right after [Summable.tendsto_cofinite_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/InfiniteSum/Group.html#Summable.tendsto_cofinite_zero)